src/views/catalog/customize/components/PersistentVolumeClaimSelect/PersistentVolumeClaimSelect.tsx

```diff
@@ -0,0 +1,146 @@
+import * as React from 'react';
+
+import { PersistentVolumeClaimModel, ProjectModel } from '@kubevirt-ui/kubevirt-api/console';
+import { V1alpha1PersistentVolumeClaim } from '@kubevirt-ui/kubevirt-api/kubevirt';
+import { useKubevirtTranslation } from '@kubevirt-utils/hooks/useKubevirtTranslation';
+import { K8sResourceCommon, useK8sWatchResource } from '@openshift-console/dynamic-plugin-sdk';
+import { FormGroup, Select, SelectOption, SelectVariant, Skeleton } from '@patternfly/react-core';
+
+import './PersistentVolumeClaimSelect.scss';
+
+type PersistentVolumeClaimSelectProps = {
+  pvcNameSelected: string;
+  namespaceSelected: string;
+  selectNamespace: (namespace: string) => void;
+  selectPVCName: (pvcName: string) => void;
+};
+
+export const PersistentVolumeClaimSelect: React.FC<PersistentVolumeClaimSelectProps> = ({
+  pvcNameSelected,
+  namespaceSelected,
+  selectPVCName,
+  selectNamespace,
+}) => {
+  const { t } = useKubevirtTranslation();
+
+  const [pvcSelectOpen, setPVCSelectOpen] = React.useState(false);
+
+  const [isNamespacePVCOpen, setNamespaceOpen] = React.useState(false);
+
+  const [projects, projectsLoaded] = useK8sWatchResource<K8sResourceCommon[]>({
+    groupVersionKind: {
+      group: ProjectModel.apiGroup,
+      version: ProjectModel.apiVersion,
+      kind: ProjectModel.kind,
+    },
+    namespaced: false,
+    isList: true,
+  });
+
+  const [pvcOptions, pvcsLoaded] = useK8sWatchResource<V1alpha1PersistentVolumeClaim[]>({
+    groupVersionKind: {
+      version: PersistentVolumeClaimModel.apiVersion,
+      kind: PersistentVolumeClaimModel.kind,
+    },
+    namespaced: false,
+    isList: true,
+  });
+
+  const projectsNames = projects.map((project) => project.metadata.name);
+
+  const pvcNamesFilteredByProjects = pvcOptions
+    .filter((pvc) => pvc.metadata.namespace === namespaceSelected)
+    .map((pvc) => pvc.metadata.name);
+
+  const onSelectNamespace = React.useCallback(
+    (event, selection) => {
+      selectNamespace(selection);
+      selectPVCName(undefined);
+      setNamespaceOpen(false);
+    },
+    [selectNamespace, selectPVCName],
+  );
+
+  const onPVCSelected = React.useCallback(
+    (event, selection) => {
+      selectPVCName(selection);
+      setPVCSelectOpen(false);
+    },
+    [selectPVCName],
+  );
+
+  const filter = (options) => {
```

Comment:
why are you creating a filter? isnt this a built in option in the `Select` component?

Reply:
It seems like you have to build your own filter function: https://www.patternfly.org/v4/components/select#grouped-single-with-filtering